flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -48,31 +96,132 @@ func ApplyInterruptibleNodeSelectorRequirement(interruptible bool, affinity *v1.
 		nodeSelectorRequirement = *config.GetK8sPluginConfig().NonInterruptibleNodeSelectorRequirement
 	}
 
-	if affinity.NodeAffinity == nil {
-		affinity.NodeAffinity = &v1.NodeAffinity{}
-	}
-	if affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
-		affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &v1.NodeSelector{}
+	AddRequiredNodeSelectorRequirements(affinity, nodeSelectorRequirement)
+}
+
+// ApplyInterruptibleNodeAffinity configures the node-affinity for the pod using the configuration specified.
+func ApplyInterruptibleNodeAffinity(interruptible bool, podSpec *v1.PodSpec) {
+	if podSpec.Affinity == nil {
+		podSpec.Affinity = &v1.Affinity{}
 	}
-	if len(affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms) > 0 {
-		nodeSelectorTerms := affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
-		for i := range nodeSelectorTerms {
-			nst := &nodeSelectorTerms[i]
-			nst.MatchExpressions = append(nst.MatchExpressions, nodeSelectorRequirement)
-		}
+
+	ApplyInterruptibleNodeSelectorRequirement(interruptible, podSpec.Affinity)
+}
+
+// Specialized merging of overrides into a base *core.ExtendedResources object. Note
+// that doing a nested merge may not be the intended behavior all the time, so we
+// handle each field separately here.
+func applyExtendedResourcesOverrides(base, overrides *core.ExtendedResources) *core.ExtendedResources {
+	// Handle case where base might be nil
+	var new *core.ExtendedResources
+	if base == nil {
+		new = &core.ExtendedResources{}
 	} else {
-		affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = []v1.NodeSelectorTerm{{MatchExpressions: []v1.NodeSelectorRequirement{nodeSelectorRequirement}}}
+		new = proto.Clone(base).(*core.ExtendedResources)
+	}
+
+	// No overrides found
+	if overrides == nil {
+		return new
+	}
+
+	// GPU Accelerator
+	if overrides.GetGpuAccelerator() != nil {
+		new.GpuAccelerator = overrides.GetGpuAccelerator()
 	}
 
+	return new
 }
 
-// ApplyInterruptibleNodeAffinity configures the node-affinity for the pod using the configuration specified.
-func ApplyInterruptibleNodeAffinity(interruptible bool, podSpec *v1.PodSpec) {
+func ApplyGPUNodeSelectors(podSpec *v1.PodSpec, gpuAccelerator *core.GPUAccelerator) {
+	// Short circuit if pod spec does not contain any containers that use GPUs
+	gpuResourceName := config.GetK8sPluginConfig().GpuResourceName
```

Comment:
I could swear I left a comment about this but don't see it... can we avoid calling `config.GetK8sPluginConfig()`... 
Can we call it once in the "root function" and just pass it down to all of these... it makes these functions more "obviously" testable...

Reply:
Following this pattern here: https://github.com/flyteorg/flyte/blob/4a780c30809d713a8e03417f6ffbc6674960bbc1/flyteplugins/go/tasks/pluginmachinery/flytek8s/container_helper.go#L167

It would be a non-trivial refactor pass config down from root via args. Happy to do it, but should that precede this PR?